arduino-core/src/processing/app/Sketch.java

```diff
@@ -351,13 +351,22 @@ public void saveAs(File newFolder) throws IOException {
         file.saveAs(new File(newFolder, file.getFileName()));
     }
 
-    folder = newFolder;
 
     // Copy the data folder (this may take a while.. add progress bar?)
     if (getDataFolder().exists()) {
       File newDataFolder = new File(newFolder, "data");
+      // Create the data folder
+      if (!newDataFolder.mkdirs()) {
```

Comment:
Should we also check for `newDataFolder.exists()` before bailing out? The check could be probably turned into `if(!(newDataFolder.exists() || newDataFolder.mkdirs()))` . Do you see any drawback in this approach?